features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/SessionReplayPrivacy.kt

```diff
@@ -79,11 +81,20 @@ enum class SessionReplayPrivacy {
     @Suppress("LongMethod")
     internal fun mappers(
         applicationId: String,
-        recordedDataQueueHandler: RecordedDataQueueHandler
+        recordedDataQueueHandler: RecordedDataQueueHandler,
+        logger: InternalLogger
     ): List<MapperTypeWrapper> {
-        val resourcesSerializer = buildResourcesSerializer(applicationId, recordedDataQueueHandler)
-        val imageWireframeHelper = ImageWireframeHelper(resourcesSerializer = resourcesSerializer)
+        val webPImageCompression = WebPImageCompression(logger)
+        val resourcesSerializer = buildResourcesSerializer(
+            applicationId,
+            recordedDataQueueHandler,
+            webPImageCompression
+        )
         val uniqueIdentifierGenerator = UniqueIdentifierGenerator
+        val imageWireframeHelper = ImageWireframeHelper(
+            resourceResolver = resourcesSerializer,
+            imageCompression = webPImageCompression
+        )
```

Comment:
Why do we include `webPImageCompression` into serializer and also into `ImageWireframeHelper`, which already has serializer included. Seems like some flaw in the design/responsibility leak (I guess it is needed only at the serialization step?).

Reply:
In the imageWireframeHelper we're using it to get the mimetype for the wireframe, and in the serializer to compress the image

Reply:
Shouldn't then serializer return MIME type with the piece serialized? Not a big deal though, just thinking.